Shared/ViewModels/AdminDashboard/DevicesViewModel.swift

```diff
@@ -12,136 +12,78 @@ import JellyfinAPI
 import OrderedCollections
 import SwiftUI
 
-final class DevicesViewModel: ViewModel, Eventful, Stateful {
+@MainActor
+@Stateful
+final class DevicesViewModel: ViewModel {
 
-    // MARK: Event
-
-    enum Event {
-        case error(JellyfinAPIError)
-        case success
-    }
-
-    // MARK: - Action
-
-    enum Action: Equatable {
+    @CasePathable
+    enum Action {
         case refresh
         case delete(ids: [String])
+        case update(id: String, options: DeviceOptionsDto)
+
+        var transition: Transition {
+            switch self {
+            case .refresh:
+                .loop(.refreshing, whenBackground: .refreshing)
+            case .delete:
+                .background(.deleting)
+            case .update:
+                .background(.updating)
+            }
+        }
     }
 
-    // MARK: - BackgroundState
-
-    enum BackgroundState: Hashable {
-        case refreshing
-        case updating
+    enum BackgroundState {
         case deleting
```

Comment:
The `deleting` background state doesn't seem to have any UI components and can be removed.